scripts/plot.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+"""plot.py
+
+Usage:
+  plot.py -a ALGORITHM [-b BACKENDS] -g GEOMETRY -i INPUT... [-n] [-o OUTPUT_FILE]
+  plot.py (-h | --help)
+
+Options:
+  -h --help                         Show this screen.
+  -a ALGORITHM --algo ALGORITHM     Algorithm ['construction', 'knn_search', 'radius_search']
+  -b BACKENDS --backends BACKENDS   Backends to parse [default: all]
+  -g GEOMETRY --geometry GEOMETRY   Geometry (source cloud) ['filled_box', 'hollow_box']
+  -i FILE --input-files=FILE        Input file containing benchmark results
+  -n --numbers                      Print numbers
+  -o FILE --output-file=FILE        Output file
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+import math
+import re
+from docopt import docopt
+
+from benchmark import parse_benchmark
+
+def libraries_comparison_rate_figure(input_files, algorithm, geometry, numbers, backend_list):
+    # Phase 0: parse backends
+    if backend_list[0] == "all":
+        backends = set()
+        for input_file in input_files:
+            with open(input_file) as f:
+                for line in f:
+                    search_result = re.search("[^<]*<ArborX::BVH<([^>]*)>.*", line)
+                    if search_result != None:
+                        backends.add(search_result.group(1))
+        backends = list(backends)
+        backends.sort()
+    else:
+        backends = backend_list
+
+    # Phase 1: data setup
+    num_sources  = []
+    timings = []
+    errors = []
+    unified_num_sources = [];
+    for i in range(len(backends)):
+        tmp_num_sources = []
+        tmp_timings = []
+        tmp_errors = []
+        for input_file in input_files:
+            tmp_file_num_sources, _, tmp_file_timings = parse_benchmark(input_file, algorithm, 'ArborX::BVH<'+backends[i]+'>', 'median', geometry)
+            tmp_file_num_sources, _, tmp_file_errors = parse_benchmark(input_file, algorithm, 'ArborX::BVH<'+backends[i]+'>', 'stddev', geometry)
+            tmp_num_sources = tmp_num_sources + tmp_file_num_sources
+            tmp_timings = tmp_timings + tmp_file_timings
+            tmp_errors = tmp_errors + tmp_file_errors
+        num_sources.append(tmp_num_sources)
+        timings.append(tmp_timings)
+        errors.append(tmp_errors)
+        unified_num_sources = unified_num_sources + tmp_num_sources
+
+    unified_num_sources = list(set(unified_num_sources))
+    unified_num_sources.sort()
+    num_source_labels = [None] * len(unified_num_sources)
+    for i in range(len(unified_num_sources)):
+        num_source_labels[i] = '$10^' + str(round(math.log(unified_num_sources[i], 10))) + '$'
```

Comment:
I'm struggling to understand what this does. Could you elaborate on what the end result of `num_sources`, `timings` or `errors` is going to be if there are repetitions in multiple files?

Reply:
If there are repetitions only the last run for a specific setting is plotted. This is actually not related to the code above but rather to how we initialize `data` below.